src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectNative.cs

```diff
@@ -256,8 +257,32 @@ internal override void ClearAllWritePackets()
             }
         }
 
-        internal override void SetPacketData(object packet, byte[] buffer, int bytesUsed) 
-            => SNINativeMethodWrapper.SNIPacketSetData((SNIPacket)packet, buffer, bytesUsed);
+        internal override void SetPacketData(object packet, byte[] buffer, int bytesUsed, SecureString[] securePasswords, int[] securePasswordOffsets)
+        {
+            if (securePasswords != null)
```

Comment:
This code is common to TdsParserStateObjectNative and TdsParserStateObjectManaged, except for the last line of function which calls different code paths. 
Can this be moved to TdsParserStateObject.cs?